rfcs/experimental/flow_graph_wait_try_put_and_wait/README.md

```diff
@@ -0,0 +1,604 @@
+
+Each input message is pre-processed in the serial ``pre_process`` node and the output is broadcasted to two concurrent computational functions
+``f1`` and ``f2``.
+
+The result is joined into a single tuple in ``join`` node and then post-processed in a serial ``post_process`` functional node.
+
+The order of processing the tasks corresponding to each input item is exposed under each node in the picture. 
+
+The tasks are shown in the right-to-left order - the tasks on the right are submitted earlier then the tasks on the left.
+
+Blue tasks relates to the parallel loop 0-100. Red tasks relates to the ``444`` message submitted as an input to ``try_put_and_wait``.
+
+Since ``pre_process`` is a serial ``function_node``, the items processing cannot be re-ordered and the red task would be in the leftmost position
+since it was submitted after the blue tasks.
+
+``f1`` and ``f2`` are ``unlimited`` ``function_node``s, so the tasks ordering is arbitrary since tasks for each computed item are spawned.
+
+``join`` is a queueing ``join_node`` and it should preserve the ordering of items as they are processed by ``f1`` and ``f2``. The tasks are shown
+as partially blue and partially red since the output of red task from ``f1`` can be combined with the output of blue task from ``f2`` and wise versa.
+
+Since ``post_process`` is a serial ``function_node``, the ordering of the tasks would be the same as in ``join``.
+
+The ``try_put_and_wait`` is expected to exit when all of the red (including partially red) tasks are completed. It may require some amount of blue tasks
+to be completed as well, e.g. to execute previously stored tasks in the ``function_node``'s queue. 
+
+Completion of all blue tasks as in ``wait_for_all`` is not guaranteed.
+
+## Feature Design
+
+This feature is implemented by creating an instance of special private class ``message_metainfo`` for each input message in ``try_put_and_wait``. This instance
+wraps the pointer to the dedicated ``wait_context`` object representing the reference counter for single input message.
+
+``message_metainfo`` is broadcasted through the graph together with the message itself. The actual value and the type of the message can be changed during the computation, but the stored 
+metainformation is preserved.
+
+The reference counter in the ``wait_context`` inside of the ``message_metainfo`` is increased in the following cases:
+* The task associated with the computations of the ``try_put_and_wait`` input message is created (*).
+* The item associated with the computations of the desired input is stored in the buffering node or in the internal buffer of the other node type.
+* When the ``continue_msg`` with the non-empty associated waiter is received by ``continue_node``. In this case, the node buffers the ``message_metainfo``s received from each
+  predecessor and increases the underlying reference counter to prolonge the wait until signals from each predecessor would be received. 
```

Comment:
Here and in other cases below where you describe the design of the already provided experimental functionality, the use of `would [be]` is somewhat confusing. Please take a look at these places and change the wording to be more definite where needed.

Reply:
Done